server/http.go

```diff
@@ -77,9 +83,5 @@ func (s *Server) logRequest(handler http.Handler) http.Handler {
 }
 
 func (s *Server) Run() error {
-	return s.https.ListenAndServe()
-}
-
-func (s *Server) Close() error {
-	return s.https.Close()
+	return s.ListenAndServe()
```

Comment:
why was this needed, with serverPrivate?

Reply:
...also why is Close removed?

Reply:
Oh I see in the commit comment, sorry

Reply:
There's no `(*http.Server).Close()` function in go1.7. I noticed that `(*server.Server).Close()` is not used anywhere in `trezord-go`. And didn't know do you need this `Close()` method or not, so I developed a solution that preserves it only for >=go1.8 and removes for <go1.8

To do that I just used unnamed embedded structure (to passthrough all methods of `*http.Server` to `server.Server`). If `*http.Server` has implemented `Close()` (>=go1.8) it will be passthrough'ed to `server.Server`. If not then won't.

But if I use directly `*http.Server` within `Server` it won't be public while `https` was public. So to preserve this property I made an intermediate structure `serverPrivate` to make `*http.Server` to be private.

So `Close()` is removed in this patch because it's already implemented in `*http.Server` and passthrough'ed to `Server`.

Reply:
Ah, the plan was to use Close() in defer in main, but we forgot to do that :D thanks for indirectly noticing

Reply:
If you need `Close()` I can prepare a new Pull Request to fix the building (after you add `Close()` into `main`).

Reply:
Hm, but now I cannot add `Close()` to be deferred in main, since I don't know whether it is or isn't go1.7, right. There are some ways I can find to do that, but they all seem overly complex.

I guess this is fine for now since exiting the app stops the server anyway.

Reply:
> Hm, but now I cannot add Close() to be deferred in main, since I don't know whether it is or isn't go1.7, right

You can add file (let's call it for example `http_go1.7.go`) with a content like (writing in blind, didn't check this):
```
// +build !go1.8
func (s *Server) Close() error {
    return s.Shutdown()
}
```

Or a dirty hack :) :
```
type serverCloser interface {
    Close() error
}
func (s *Server) Close() error {
    closer, ok := (interface{})((*http.Server)(s)).(serverCloser)
    if ok {
        return closer.Close()
    }
    return s.Shutdown()
}
```

Reply:
Ok that all looks too hacky :) this is fine for now, since it is not a regression anyway, the server closes on exit fine

Reply:
In my very humble opinion, this looks nice — `server/http_go-lt1.8.go`:
```go
// +build !go1.8

func (s *Server) Close() error {
    return s.Shutdown()
}
```

Anyway. Thanks again for the merging)

Reply:
Shutdown() doesn't seem to work on 1.7 either

Reply:
Ok. I'll research for another way.

Reply:
Here's an implementation of Close(): https://github.com/trezor/trezord-go/pull/82